csv.h

```diff
@@ -1142,8 +1144,8 @@ class CSVReader {
  private:
   void parse_helper(std::size_t) {}
 
-  template <class T, class... ColType>
-  void parse_helper(std::size_t r, T &t, ColType &... cols) {
+  template <class T, class... optional<ColType>>
+  void parse_helper(std::size_t r, T &t, optional<ColType> &... cols) {
```

Comment:
I think you want this instead

```suggestion
  template <class T, class... ColType>
  void parse_helper(std::size_t r, T &t, optional<ColType> &... cols) {
```